### ISSUE TITLE
edgehandles canvas not overlapping cytoscape layers

### ISSUE THREAD

**Issue by hauswirth** [cytoscape/cytoscape.js-edgehandles#28]
I played around with edgehandles and couldn't see the handles (everything worked, just no handles were drawn on the node). Eventually I discovered that the handles do appear on my web page, but outside the cytoscape layer canvases.

Edgehandles creates its own canvas, which it tries to place exactly over the cytoscape canvases. It injects its canvas into the DOM, as follows:

```javascript
$container.append( $canvas );
```

For me, this leads to the canvas showing up below (in y coordinates) the cy element (and its layer canvases), instead of overlapping the cy element.

The reason seems to be that edgehandles appends the canvas to the container DIV, instead of the the canvas container DIV.

Changing the above line in *cytoscape-edgehandles.js* into this:

```javascript
$container.find('.canvas-container').append($canvas);
```

**together** with a corresponding change in the `CanvasRenderer` function in *cytoscape.js* (to add the 'canvas-container' class to the canvas container DIV) should fix this:

```javascript
  r.data.canvasContainer.setAttribute('class', 'canvas-container');
```

**Comment by maxkfranz** [comment]
Thanks for the feedback.  I pushed some changes that should resolve this.

I don't want to take your code suggestion directly, because that would involve tighter coupling with the renderer.  The change I made should also be backwards compatible with older versions of Cy.js, and 
I think that's preferable.

Try it out, and let me know what you think

**Comment by maxkfranz** [comment]
Alternative solution: Prepend the div with absolute positioning.  This approach seems to work well for the context menu extension:

https://github.com/cytoscape/cytoscape.js-cxtmenu

**Comment by moklick** [comment]
Hey,

we just tried the edgehandles plugin and we also can't see any handles. It's also not possible to drag the nodes anymore if we activate the plugin.

**Comment by maxkfranz** [comment]
@moklick Did you try the latest master with the fix?  2.2.1 hasn't been released yet.  If so, please provide your browser and OS (versions etc).

@hauswirth Any updates on whether the fix works for you?

**Comment by maxkfranz** [comment]
I've just decided to push the change as a release.  If there are problems with it, we can create another issue for 2.2.2
